tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+* @file test_signal_generator.cpp
+* @brief This file creates an unit test for checking the main functionalities of the 
+* SignalGeneratorComponent and the DigitalConverteComponent
+*
+* @author Murilo Mendonça Venâncio
+* 
+* @date 24/01/2019
+*/
+
+#include <hyro/SignalGeneratorComponent.hpp>
+#include <hyro/DigitalConverterComponent.hpp>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <hyro/test/ComponentTester.h>
+#include <gtest/gtest.h>
+
+static void
+__assert_and_run_state_machine(hyro::StateMachine& sm,
+                               const hyro::ComponentConfiguration& conf,
+                               const hyro::ConnectionConfiguration& endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+#define ASSERT_AND_RUN_STATE_MACHINE(sm, conf, endpoint) \
+  __assert_and_run_state_machine(sm, conf, endpoint)
+
+
+
+namespace hyro
+{
+namespace test
+{
+
+TEST (TestSignalGenerator, SignalGeneratorComponentCheck)
```

Comment:
Your tests are very well programmed. However, I would check the value of the signals output channel after changing the dynamic parameters.